src/ApiManagement/ApiManagement.ServiceManagement/Commands/GetAzureApiManagementProduct.cs

```diff
@@ -49,6 +50,13 @@ public class GetAzureApiManagementProduct : AzureApiManagementCmdletBase
             HelpMessage = "Title of the Product to look for. If specified will try to get the Product by title. This parameter is optional.")]
         public String Title { get; set; }
 
+        [Parameter(
+            ParameterSetName = GetByApiId,
+            ValueFromPipelineByPropertyName = true,
+            Mandatory = false,
```

Comment:
@solankisamir if this is the only parameter in this set (aside from the context), then we should make it mandatory